### DISCUSSION TITLE
Loading a large collections with livewire powergrid

### DISCUSSION THREAD

**Discussion by DalitsoMandala** [Power-Components/livewire-powergrid#1666] [Q&A] [answered]
When using Eloquent Query and loading 1million records it works perfectly but when i use collections it takes forever to load. How can i optimize this without having any memory issues?

**Comment by luanfreitasdev** [comment, marked as answer]
Hello!

Using a collection with a database with so many records is not advisable. The performance problem will depend on how many columns you are displaying, and the database latency, and with 1 million records in the collection, it will overflow the memory.

**Reply by DalitsoMandala** [reply]
Alright thanks
